### ISSUE TITLE
Support for RainbowKit 0.12.x

### ISSUE THREAD

**Issue by fodisi** [celo-org/rainbowkit-celo#44]
Hi.
Which version should be used with rainbowkit 0.12.x? this rainbowkit version has support for wallet connect v2, without the need to fully migrate to v1.x.x. But it still accepts passing `projectId` from WC Cloud, which is now required.

Thanks

**Comment by aaronmgdr** [comment]
ah this version slipped by us.

**Comment by aaronmgdr** [comment]
having more trouble with this than the 1.0 release. 🫤

**Comment by fodisi** [comment]
> having more trouble with this than the 1.0 release. 🫤

Oh no - sorry to hear that.

Have you've seen/used the newly released [ethers.js adapters](https://wagmi.sh/react/ethers-adapters) for wagmi 1.x.x? This allows projects to continue using ethers.js w/ wagmi 1.x, and incrementally migrate to viem.
I believe celo-rainbowkit v1 would also work with these adpaters without issues, correct? Or would it require further changes?

Thanks for all the work (and fighting all the trouble ;-) ) to get this released, @aaronmgdr .

**Comment by dckesler** [comment]
@fodisi Yes, you should be able to follow the tutorials that you linked in your comment above for your application without any changes to rainbowkit-celo.

I was able to get it working on our example in this repo and I've made a [branch](https://github.com/celo-org/rainbowkit-celo/tree/ethers-example) for reference. You can see [here](https://github.com/celo-org/rainbowkit-celo/blob/ethers-example/apps/example/utils/use-ethers.ts) where I implemented the code given by the wagmi team. Then you can see [here](https://github.com/celo-org/rainbowkit-celo/blob/ethers-example/apps/example/pages/ethers.tsx) where I use both the ethers signer and the ethers provider.

Hopefully, this is also sufficient for your needs. Please reach out again if you run into any issues!

**Comment by dckesler** [comment]
Also, here's a more in-depth tutorial I posted to Celo Academy that includes a guide on using Ethers with rainbowkit/wagmi 1.x.x.
https://celo.academy/t/upgrading-your-dapp-to-wallet-connect-v2/2907?u=daniel_kesler
